src/common.jl

```diff
@@ -44,8 +60,60 @@ struct Interval
     right::Float64
 end
 
+union(i1::Interval, i2::Interval) =
+    Interval(min(i1.left, i2.left), max(i1.right, i2.right))
+
 function Base.read(io::IO, ::Type{Interval})
     left = read(io, Float64)
     right = read(io, Float64)
     Interval(left, right)
 end
+
+function Base.write(io::IO, interval::Interval)
+    Base.write(io, interval.left, interval.right)
+end
+
+"""
+    Header
+
+Common header read/write object for shp and shx files.
+"""
+struct Header
+    code::Int32
+    filesize::Int32
+    version::Int32
+    shapecode::Int32
+    MBR::Rect
+    zrange::Interval
+    mrange::Interval
+end
+function Header(; filesize, shapecode, mbr, zrange, mrange)
+    code = Int32(9994)
```

Comment:
Can use a comment about the header structure/link to spec. Magicbytes I assume, normally displayed as unsigned?